aiosmtpd/smtp.py

```diff
@@ -317,21 +321,27 @@ def _set_rset_state(self):
                 # The connection got reset during the DATA command.
                 # XXX If handler method raises ConnectionResetError, we should
                 # verify that it was actually self._reader that was reset.
-                log.info('Connection lost during _handle_client()')
+                log.info('%r Connection lost during _handle_client()',
+                         self.session.peer)
                 self._writer.close()
                 raise
             except Exception as error:
                 try:
                     status = await self.handle_exception(error)
-                    await self.push(status)
-                except Exception as error:
+                except Exception as error2:
                     try:
-                        log.exception('Exception in handle_exception()')
+                        log.exception('%r Exception in handle_exception()',
+                                      self.session.peer)
                         status = '500 Error: ({}) {}'.format(
-                            error.__class__.__name__, str(error))
+                            error2.__class__.__name__, str(error2))
                     except Exception:
                         status = '500 Error: Cannot describe error'
-                    await self.push(status)
+                finally:
+                    if isinstance(error, TLSSetupException):
```

Comment:
Forgive me if I've missed something - but I don't see `error` defined other than in the exception. An example might help:

```
>>> try:
...   print('hi')
... except Exception as error:
...   ...
... finally:
...   print('error?', error)
... 
hi
Traceback (most recent call last):
  File "<stdin>", line 6, in <module>
NameError: name 'error' is not defined
```

If the code is successful, won't this fail with `NameError`?

Reply:
I cannot see my comment anymore.  
I is defined in the main exception handler (line 328)